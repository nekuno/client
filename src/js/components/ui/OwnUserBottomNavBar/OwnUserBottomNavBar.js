import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './OwnUserBottomNavBar.scss';
import translate from "../../../i18n/Translate";

@translate('OwnUserBottomNavBar')
export default class OwnUserBottomNavBar extends Component {

    static propTypes = {
        current      : PropTypes.oneOf(['about-me', 'networks', 'friends', 'answers', 'interests']),
        notifications: PropTypes.number,
    };

    static contextTypes = {
        router: PropTypes.object.isRequired
    };

    constructor(props) {
        super(props);

        this.handleClickAboutMe = this.handleClickAboutMe.bind(this);
        this.handleClickNetworks = this.handleClickNetworks.bind(this);
        this.handleClickFriends = this.handleClickFriends.bind(this);
        this.handleClickAnswers = this.handleClickAnswers.bind(this);
        this.handleClickInterests = this.handleClickInterests.bind(this);
        this.getIconClassname = this.getIconClassname.bind(this);
    }

    handleClickAboutMe() {
        this.redirectTo('about-me');
    }

    handleClickNetworks() {
        this.redirectTo('networks');
    }

    handleClickFriends() {
        this.redirectTo('friends');
    }

    handleClickAnswers() {
        this.redirectTo('answers');
    }

    handleClickInterests() {
        this.redirectTo('interests');
    }

    redirectTo(route) {
        const {current} = this.props;

        if (current !== route) {
            this.context.router.push('/' + route);
        }
    }

    getIconClassname(section) {
        return this.props.current === section ? styles.iconWrapper + ' ' + styles.current : styles.iconWrapper;
    }

    render() {
        const {notifications, strings} = this.props;

        return (
            <div className={styles.ownUserBottomNavbar}>
                <div className={this.getIconClassname('about-me')} onClick={this.handleClickAboutMe}>
                    <div className={styles.icon + ' icon icon-user'}/>
                    <div className={styles.iconText + ' small'}>{strings.aboutMe}</div>
                </div>
                <div className={this.getIconClassname('networks')} onClick={this.handleClickNetworks}>
                    <div className={styles.icon + ' icon icon-share-2'}/>
                    <div className={styles.iconText + ' small'}>{strings.networks}</div>
                </div>
                <div className={this.getIconClassname('friends')} onClick={this.handleClickFriends}>
                    <div className={styles.icon + ' icon icon-copy'}/>
                    <div className={styles.iconText + ' small'}>{strings.friends}</div>
                </div>
                <div className={this.getIconClassname('answers')} onClick={this.handleClickAnswers}>
                    <div className={styles.icon + ' icon icon-check-square'}/>
                    <div className={styles.iconText + ' small'}>{strings.answers}</div>
                </div>
                <div className={this.getIconClassname('interests')} onClick={this.handleClickInterests}>
                    <div className={styles.icon + ' icon icon-bookmark'}/>
                    {notifications ?
                        <div className={styles.notificationsWrapper}>
                            <div className={styles.notifications}>
                                {notifications > 99 ? 99 : notifications}
                            </div>
                        </div>
                        : null
                    }
                    <div className={styles.iconText + ' small'}>{strings.interests}</div>
                </div>
            </div>
        );
    }
}

OwnUserBottomNavBar.defaultProps = {
    strings: {
        aboutMe  : 'About me',
        networks : 'Networks',
        friends  : 'Friends',
        answers  : 'Answers',
        interests: 'Interests',
    }
};