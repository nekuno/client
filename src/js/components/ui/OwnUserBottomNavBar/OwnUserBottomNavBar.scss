@import '../../../../scss/_partials/colors';

.own-user-bottom-navbar {
  background: $white-background;
  box-shadow: 0 -2px 5px $gray-border;
  bottom: 0;
  color: $gray-main-text;
  font-size: 14px;
  height: 50px;
  line-height: 50px;
  position: sticky;
  width: 100%;

  .icon-wrapper {
    color: $gray-main-text;
    cursor: pointer;
    display: inline-block;
    height: 50px;
    opacity: 0.5;
    text-align: center;
    vertical-align: middle;
    width: 20%;

    .icon {
      font-size: 28px;
      width: 100%;
    }
    .icon-text {
      font-size: 10px;
      width: 100%;
    }

    &.current {
      cursor: default;
      opacity: 1;
    }

    .notifications-wrapper {
      background: $green-experiences;
      border-radius: 50%;
      -moz-border-radius: 50%;
      -webkit-border-radius: 50%;
      position: absolute;
      text-align: center;
      line-height: 20px;
      margin-left: 11%;
      top: 15px;
      height: 20px;
      width: 20px;

      .notifications {
        color: #fff;
        line-height: 20px;
        font-size: 10px;
        font-weight: bold;
      }
    }
  }
}