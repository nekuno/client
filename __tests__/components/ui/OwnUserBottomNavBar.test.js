import React from 'react';
import OwnUserBottomNavBar from '../../../src/js/components/ui/OwnUserBottomNavBar/OwnUserBottomNavBar.js';
import styles from '../../../src/js/components/ui/OwnUserBottomNavBar/OwnUserBottomNavBar.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test OwnUserBottomNavBar component', () => {
    const mockFn = jest.fn();
    const ownuserbottomnavbar = shallow(<OwnUserBottomNavBar onClickHandler={mockFn}>Lorem ipsum</OwnUserBottomNavBar>);

    it('should be defined', () => {
        expect(ownuserbottomnavbar).toBeDefined();
    });
    it('should render correctly', () => {
        expect(ownuserbottomnavbar).toMatchSnapshot();
    });
    it('OwnUserBottomNavBar click event', () => {
        ownuserbottomnavbar.find(`.${styles.ownuserbottomnavbar}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
