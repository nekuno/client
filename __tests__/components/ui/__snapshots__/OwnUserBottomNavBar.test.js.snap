// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test OwnUserBottomNavBar component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranslationComponent
    onClickHandler={[MockFunction]}
    strings={
      Object {
        "aboutMe": "About me",
        "answers": "Answers",
        "friends": "Friends",
        "interests": "Interests",
        "networks": "Networks",
      }
    }
  >
    Lorem ipsum
  </TranslationComponent>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Lorem ipsum",
      "locale": undefined,
      "onClickHandler": [MockFunction],
      "strings": Object {
        "aboutMe": "About me",
        "answers": "Answers",
        "friends": "Friends",
        "interests": "Interests",
        "networks": "Networks",
      },
    },
    "ref": null,
    "rendered": "Lorem ipsum",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Lorem ipsum",
        "locale": undefined,
        "onClickHandler": [MockFunction],
        "strings": Object {
          "aboutMe": "About me",
          "answers": "Answers",
          "friends": "Friends",
          "interests": "Interests",
          "networks": "Networks",
        },
      },
      "ref": null,
      "rendered": "Lorem ipsum",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
